### ISSUE TITLE
Teething troubles

### ISSUE THREAD

**Issue by topiaruss** [simplyzaki/docker-images#1]
Hello sir, may I call you Zak?

I've been experimenting with precise-cassandra-titan as pulled from the Docker index. 

I run with:

```
sudo docker run -d khanio/precise-cassandra-titan
```

And the container shows up with docker ps.

I had trouble connecting so looked at the logs:

```
root@602806c13c3f:/# 2013-11-03 22:36:39,672 CRIT Supervisor running as root (no user in config file)
2013-11-03 22:36:39,673 WARN Included extra file "/etc/supervisor/conf.d/rexster-server.conf" during parsing
2013-11-03 22:36:39,673 WARN Included extra file "/etc/supervisor/conf.d/elasticsearch.conf" during parsing
2013-11-03 22:36:39,673 WARN Included extra file "/etc/supervisor/conf.d/cassandra.conf" during parsing
2013-11-03 22:36:39,673 WARN Included extra file "/etc/supervisor/conf.d/sshd.conf" during parsing
2013-11-03 22:36:39,699 INFO RPC interface 'supervisor' initialized
2013-11-03 22:36:39,700 WARN cElementTree not installed, using slower XML parser for XML-RPC
2013-11-03 22:36:39,700 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2013-11-03 22:36:39,701 INFO supervisord started with pid 15
2013-11-03 22:36:40,708 INFO spawned: 'rexster-server' with pid 16
2013-11-03 22:36:40,709 INFO spawned: 'sshd' with pid 17
2013-11-03 22:36:40,711 INFO spawned: 'elasticsearch' with pid 18
2013-11-03 22:36:40,714 INFO spawned: 'cassandra' with pid 19
2013-11-03 22:36:40,750 INFO exited: rexster-server (exit status 0; not expected)
2013-11-03 22:36:40,760 INFO exited: elasticsearch (exit status 0; not expected)
2013-11-03 22:36:40,960 INFO exited: cassandra (exit status 0; not expected)
2013-11-03 22:36:41,966 INFO spawned: 'rexster-server' with pid 168
2013-11-03 22:36:41,966 INFO success: sshd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2013-11-03 22:36:41,968 INFO spawned: 'elasticsearch' with pid 169
2013-11-03 22:36:41,970 INFO spawned: 'cassandra' with pid 170
2013-11-03 22:36:42,004 INFO exited: rexster-server (exit status 0; not expected)
2013-11-03 22:36:42,006 INFO exited: elasticsearch (exit status 0; not expected)
2013-11-03 22:36:42,340 INFO exited: cassandra (exit status 0; not expected)
2013-11-03 22:36:44,346 INFO spawned: 'rexster-server' with pid 280
2013-11-03 22:36:44,354 INFO spawned: 'elasticsearch' with pid 281
2013-11-03 22:36:44,357 INFO spawned: 'cassandra' with pid 282
2013-11-03 22:36:44,382 INFO exited: elasticsearch (exit status 0; not expected)
2013-11-03 22:36:44,412 INFO exited: rexster-server (exit status 0; not expected)
2013-11-03 22:36:44,724 INFO exited: cassandra (exit status 0; not expected)
2013-11-03 22:36:47,761 INFO spawned: 'rexster-server' with pid 524
2013-11-03 22:36:47,783 INFO spawned: 'elasticsearch' with pid 525
2013-11-03 22:36:47,791 INFO spawned: 'cassandra' with pid 526
2013-11-03 22:36:47,850 INFO exited: elasticsearch (exit status 0; not expected)
2013-11-03 22:36:47,874 INFO gave up: elasticsearch entered FATAL state, too many start retries too quickly
2013-11-03 22:36:47,911 INFO exited: rexster-server (exit status 0; not expected)
2013-11-03 22:36:48,191 INFO gave up: rexster-server entered FATAL state, too many start retries too quickly
2013-11-03 22:36:48,236 INFO exited: cassandra (exit status 0; not expected)
2013-11-03 22:36:49,242 INFO gave up: cassandra entered FATAL state, too many start retries too quickly
```

I see the same logs printed if I do not put the container in -d mode. 

Do you have any thoughts about what might be causing this?  I've gone back a second time to grab the image, and get the same behaviour.

If I can get this working, I'll be happy to use it and contribute to your Dockerfiles. It's the most comprehensive I have found, and the doc is useful. Thank you.

Best regards from a cold and rainy London,

--r

Edit:
I forgot to say that I did stop the docker service, and ran 

> sudo echo "limit nofile 262144 262144" >> /etc/init/docker.conf
> --r.

**Comment by simplyzaki** [comment]
Hi @topiaruss 

To start with, you cannot run this docker container without specifying -d, i.e the container must (always) be started in  daemon mode. Once you start in daemon mode and you see new container created using "docker ps", give it a minute for all services to start and be available. To test if all are services are up and running, you can point your browser to http://127.0.0.1:8182 to view the Rexster web interface. Hope this helps.

Cheers.

**Comment by topiaruss** [comment]
Right. Thank you for that, and for responding so fast.  I was able to access rexster immediately.

My next problem is that I see the ports in the image binding withing tcp6, not 4 (below). My test is trying to bind to tcp4 ports and failing. Clearly the browser is not having a problem at :::8182.

Edit: This issue was fixed by exactly following leseb's instructions here: 
http://serverfault.com/questions/545379/docker-will-only-bind-forwarded-ports-to-ipv6-interfaces

Thanks again.

--r

(initial_tests)vagrant@precise64:~/shipyard/initial_tests/initial_tests$ netstat -an | grep LISTEN
tcp        0      0 0.0.0.0:111             0.0.0.0:\*               LISTEN
tcp        0      0 10.0.3.1:53             0.0.0.0:\*               LISTEN
tcp        0      0 0.0.0.0:22              0.0.0.0:\*               LISTEN
tcp6       0      0 :::9160                 :::\*                    LISTEN
tcp6       0      0 :::111                  :::\*                    LISTEN
tcp6       0      0 :::9200                 :::\*                    LISTEN
tcp6       0      0 fe80::50a7:3bff:fe67:53 :::\*                    LISTEN
tcp6       0      0 :::8182                 :::\*                    LISTEN
tcp6       0      0 :::22                   :::\*                    LISTEN
